core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -231,10 +231,20 @@ public static void ddl(
     ) throws SqlException {
         CompiledQuery cc = compiler.compile(ddl, sqlExecutionContext);
         switch (cc.getType()) {
-            case INSERT:
-                throw SqlException.$(0, "use insert()");
+            case CREATE_TABLE:
+            case CREATE_TABLE_AS_SELECT:
             case DROP:
-                throw SqlException.$(0, "use drop()");
+                assert sqlExecutionContext.getCairoEngine() == compiler.getEngine();
+                try (Operation op = cc.getOperation()) {
+                    assert op != null;
+                    try (OperationFuture fut = op.execute(sqlExecutionContext, null)) {
+                        fut.await();
+                    }
+                }
+                break;
+            case INSERT:
+                insert(compiler, ddl, sqlExecutionContext);
```

Comment:
is this intentional? the method is called `ddl()`, it disallow SELECTs, but is alright with INSERT? Why?

Reply:
renamed to "execute" here: https://github.com/questdb/questdb/pull/5159/commits/ddd558e25ac0b31fd5e8ae974eebde6c68223a56